src/DependencyInjection/HttplugExtension.php

```diff
@@ -382,7 +383,14 @@ private function configureClient(ContainerBuilder $container, $clientName, array
         $serviceId = 'httplug.client.'.$clientName;
 
         if (method_exists($container, 'registerAliasForArgument')) {
-            $container->registerAliasForArgument($serviceId, HttpClient::class, $clientName);
+            $alias = $container->registerAliasForArgument($serviceId, HttpClient::class, $clientName);
+
+            $interfaces = class_implements(HttpClient::class) ?? [];
+            if (isset($interfaces[ClientInterface::class])) {
+                $alias->setDeprecated('php-http/httplug-bundle', '1.22', 'The "%alias_id%" alias is deprecated, use "Psr\Http\Client\ClientInterface" instead.');
```

Comment:
@ruudk somebody reported that symfony complains about the deprecation warning, saying it needs an `%alias_id%` (though i see that here)... https://twitter.com/PP_Vortex/status/1419599426084032519

can you please check if the deprecation warning works correctly for you and there is no symfony warning about alias_id?

Reply:
I did some deep dive into vendors and found out the following fact:

the method `setDeprecated` in Alias.php changed in symfony 5.3 compared to symfony 4.4:
https://github.com/symfony/dependency-injection/blob/5.3/Alias.php
https://github.com/symfony/dependency-injection/blob/4.4/Alias.php


In symfony 4.4, the second parameter is the template. So when using http plug bundle with symfony 4.4 executing _assets:install_ interpretes this line as "deprecation template is '1.22'" and this string obviously doesn't contain %alias_id%. 


**My conclusion: Currently, php-http/httplug-bundle 1.22 is not compatible with symfony 4.4 but seems to be compatible with symfony 5.3** 

What would be the next steps here? Would you like to be compatible with both versions?

Reply:
I'll check it out and make a PR.

Reply:
See https://github.com/php-http/HttplugBundle/pull/396

Reply:
thanks for reporting the issue and for the analysis @franziskahahn. the deprecation is removed in https://github.com/php-http/HttplugBundle/releases/tag/1.22.1